src/theory/ee_manager.cpp

```diff
@@ -41,10 +41,11 @@ eq::EqualityEngine* EqEngineManager::allocateEqualityEngine(EeSetupInfo& esi,
   if (esi.d_notify != nullptr)
   {
     return new eq::EqualityEngine(
-        *esi.d_notify, c, esi.d_name, esi.d_constantsAreTriggers);
+        d_env, c, *esi.d_notify, esi.d_name, esi.d_constantsAreTriggers);
   }
   // the theory doesn't care about explicit notifications
-  return new eq::EqualityEngine(c, esi.d_name, esi.d_constantsAreTriggers);
+  return new eq::EqualityEngine(
+      d_env, c, esi.d_name, esi.d_constantsAreTriggers);
```

Comment:
I realize that this is probably out of scope for this PR, but we should refactor it to return a `std::unique_ptr<>`.